File: abstractions/php/src/Types/Time.php
<?php
/**
 * Copyright (c) Microsoft Corporation.  All Rights Reserved.
 * Licensed under the MIT License.  See License in the project root
 * for license information.
 */

namespace Microsoft\Kiota\Abstractions\Types;

use DateTime;
use Exception;

/**
 * This object represents time in hours minutes and seconds
 */
class Time
{

    /**
     * The final string representation of the Time
     * @var string $value
     */
    private string $value;

    /**
     * @param string $timeString The time value in string format HH:MM:SS
     * H - Hour
     * M - Minutes
     * S - Seconds
     * @throws Exception
     */
    public function __construct(string $timeString) {
        $this->value = (new DateTime($timeString))->format('H:i:s');
    }

    /**
     * Creates a Time object from a DateTime object
     * @param DateTime $dateTime
     * @return Time
     * @throws Exception
     */
    public static function createFromDateTime(DateTime $dateTime): Time {
        return new self($dateTime->format('H:i:s'));
    }

    /**
     * Creates a new Time object from $hour,$minute and $seconds
     * @param int $hour
     * @param int $minutes
     * @param int $seconds
     * @return Time
     * @throws Exception
     */
    public static function createFrom(int $hour, int $minutes, int $seconds = 0): Time {
        $date = new DateTime('1970-12-12T00:00:00Z');
        $date->setTime($hour, $minutes, $seconds);
        return self::createFromDateTime($date);
    }

    /**
     * @return string
     */
    public function __toString() {
        return $this->value;
    }
}

File: abstractions/dotnet/src/Time.cs
// ------------------------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
// ------------------------------------------------------------------------------

using System;

namespace Microsoft.Kiota.Abstractions
{
    /// <summary>
    /// Model to represent only the date component of a DateTime
    /// </summary>
    public struct Time
    {
        /// <summary>
        /// Create a new Time from hours, minutes, and seconds.
        /// </summary>
        /// <param name="hour">The hour.</param>
        /// <param name="minute">The minute.</param>
        /// <param name="second">The second.</param>
        public Time(int hour, int minute, int second)
            : this(new DateTime(1, 1, 1, hour, minute, second))
        {
        }

        /// <summary>
        /// Create a new Time from a <see cref="DateTime"/> object
        /// </summary>
        /// <param name="dateTime">The <see cref="DateTime"/> object to create the object from.</param>
        public Time(DateTime dateTime)
        {
            this.DateTime = dateTime;
        }

        /// <summary>
        /// The <see cref="DateTime"/> representation of the class
        /// </summary>
        public DateTime DateTime { get; }

        /// <summary>
        /// The hour.
        /// </summary>
        public int Hour
        {
            get
            {
                return this.DateTime.Hour;
            }
        }

        /// <summary>
        /// The minute.
        /// </summary>
        public int Minute
        {
            get
            {
                return this.DateTime.Minute;
            }
        }

        /// <summary>
        /// The second.
        /// </summary>
        public int Second
        {
            get
            {
                return this.DateTime.Second;
            }
        }

        /// <summary>
        /// The time of day, formatted as "HH:mm:ss".
        /// </summary>
        /// <returns>The string time of day.</returns>
        public override string ToString()
        {
            return this.DateTime.ToString("HH:mm:ss");
        }
    }
}
